packages/frontend-shared/src/gql-components/modals/SelectCloudProjectModal.vue

```diff
@@ -317,12 +337,20 @@ const projectOptions = computed(() => {
 const newProject = ref(projectOptions.value.length === 0)
 const pickedProject = ref<typeof projectOptions.value[number]>()
 
-watchEffect(() => {
-  if (projectOptions.value.length === 1) {
+watch(projectOptions, (newVal, oldVal) => {
+  // avoid unselecting currently chosen project
+  // can happen when gql updates due to polling
+  if (isEqual(newVal, oldVal)) {
```

Comment:
Rather than check for equality of `projectOptions`, can you check if `pickedProject.value` is defined?

Reply:
In some cases we need to replace or unset the value of `pickedProject` - if you have value there and change organizations in the dropdown above, we use this watcher to clear out the now stale choice. 

Unless both orgs have the exact same project list, in which case this watcher does nothing (but the stale works in both lists, so it's ok).